packages/wp-now/src/config/ConfigFactory.ts

```diff
@@ -0,0 +1,45 @@
+import CliConfigProvider, {
+	CliOptions,
+} from './providers/CliConfigProvider';
+import { WPNowOptions } from './providers/ConfigProvider';
+import ConfigMerger from './ConfigMerger';
+import WpEnvConfigProvider from './providers/WpEnvConfigProvider';
+
+export interface ConfigFactoryOptions {
+	cliOptions?: CliOptions;
+}
+
+export interface ConfigFactoryInterface {
+	read(options: ConfigFactoryOptions): Promise<WPNowOptions>;
+}
+
+class ConfigFactory implements ConfigFactoryInterface {
```

Comment:
Singletons and factories may be a lot to take in for new contributors. This class seems to boil down to the following function type – perhaps it could be simplified?

```ts
type ConfigFactory = (options: ConfigFactoryOptions): Promise<WPNowOptions>;
```

Reply:
Indeed. I was down the OOP path! :D
Will change that to what you proposed.

Reply:
It shouldn't be a big deal – it's only read once. If it's ever a problem, `export default memo( readConfig )` should do the trick.